### PR TITLE
[BUGFIX] Fixes #4397 add silentError with deprecation

### LINKED ISSUE #4397: Installing ember-cli-deploy on ember-cli@1.13.0 fails
Steps to reproduce:

1. ember new my-app
2. ember install ember-cli-deploy
3. Boom.

```
Cannot find module 'ember-cli/lib/errors/silent'
Error: Cannot find module 'ember-cli/lib/errors/silent'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/Users/user/my-app/node_modules/ember-cli-deploy/lib/utilities/configuration-reader.js:4:19)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
```

### PR DESCRIPTION
**Pull request by trabus**
Fixes #4397 

I can try and write a quick test for this as well if desired.

### CODE REVIEW

File: lib/errors/silent.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+var silentError = require('silent-error');
+var deprecate   = require('../utilities/deprecate');
+
+module.exports = function(){
```
Comment (stefanpenner) [inline lib/errors/silent.js]: this would need to be 

```
Object.defineProperty(module, 'exports', {
  get: function () { return SilentError; }
});
```

Comment (stefanpenner) [inline lib/errors/silent.js]: although a pain, we should add a test to confirm.

Comment (trabus) [inline lib/errors/silent.js]: Okay.

File: tests/unit/errors/silent-test.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+var SilentError    = require('silent-error');
+var SilentErrorLib = require('../../../lib/errors/silent');
+var expect         = require('chai').expect;
+
+describe('SilentError', function() {
+  it('return silent-error and print a deprecation', function() {
+    expect(SilentErrorLib, 'returns silent-error').to.equal(SilentError);
```
Comment (trabus) [inline tests/unit/errors/silent-test.js]: @stefanpenner This prints a deprecation when testing, but I'm completely forgetting how I can capture that to validate.

**Pull request merged**